### DISCUSSION TITLE
Participant Email Addresses Input

### DISCUSSION THREAD

**Discussion by iamericfletcher** [lukevella/rallly#201] [Ideas]
Hi, everyone. 

I'd like to float the idea of adding a feature whereby poll creators can collect participants' emails in addition to their names. 

Perhaps a checkbox could be added during the new poll creation process? I'm thinking below the step 1 description text box could be a good spot.

If checked, the email input field would display below where participants enter their names. 

This would be _incredibly_ helpful for executive assistants like me who oftentimes create calendar invitations after the poll is complete. I have an open poll now and within the description, I ask participants to please provide their email in addition to their name (e.g. Eric Fletcher - efletch@mit.edu). Unfortunately, only a handful of the 70+ participants followed the instructions to provide their email after their name. I expected this and believe that having a dedicated field for email would help in situations like mine! 

I created a quick and dirty example demonstrating where the email input could reside in the poll. Of course, this is just a quick proof of concept! Created `email-input.tsx` in **components** and slapped it on. 

Depending on what people think, I'd be happy to create a more robust example of what I have in mind. If it's well-received, I'd be happy to have the feature assigned to me :)

<img width="778" alt="image" src="https://user-images.githubusercontent.com/64165327/173981966-cd716734-f31d-43ed-93ce-fa9dc73439cc.png">

**Comment by lukevella** [comment]
Hey Eric, thanks for this. I do have plans to support this in the future. It will be possible to request other pieces of info from participants but need a few more iterations on the UI to get there. It's quite high on the priority list so shouldn't be too long before it is available. Aiming for 3 - 4 weeks.

**Reply by iamericfletcher** [reply]
Absolutely terrific! Very glad to hear this :)

**Comment by Gamca** [comment]
Email is one of the most wanted definitely. However in order to support any other future requests, I was wondering if this could be solved more "dynamically" and there could be possibility for Admin to simply select which fields shoudl be collected from User in order to vote.
I would prob. prefer more "robust" solution giving options to select if the added field is for example text field (which length), picklist (with options), multipicklist, date, email.. so some sntadards..  + for each field specify if it is optional or mandatory.
